CMakeLists.txt

```diff
@@ -133,3 +133,48 @@ if(AMQP-CPP_LINUX_TCP)
     find_package(OpenSSL REQUIRED)
     target_include_directories(${PROJECT_NAME} PRIVATE ${OPENSSL_INCLUDE_DIR})
 endif()
+
+# software packaging
+# ------------------------------------------------------------------------------------------------------
+
+# if unix use deb and rpm generators, else default to .tar.gz
+if (UNIX)
+    set(CPACK_GENERATOR DEB RPM)
+endif (UNIX)
+
+# general cpack settings
+set(CPACK_PACKAGE_DESCRIPTION_SUMMARY "C++ library for communicating with a RabbitMQ message broker")
+set(CPACK_PACKAGE_VENDOR "Coppernica Marketing Software")
+set(CPACK_PACKAGE_CONTACT "Copernica <info@copernica.com>")
+set(CPACK_PACKAGE_DESCRIPTION
+    "AMQP-CPP is a C++ library for communicating with a RabbitMQ message broker. The
+    library can be used to parse incoming data from a RabbitMQ server, and to
+    generate frames that can be sent to a RabbitMQ server."
+)
+set(CPACK_PACKAGE_HOMEPAGE_URL "https://github.com/CopernicaMarketingSoftware/AMQP-CPP")
+set(CPACK_RESOURCE_FILE_LICENSE "${CMAKE_CURRENT_SOURCE_DIR}/LICENSE")
+set(CPACK_PACKAGE_VERSION_MAJOR ${VERSION_MAJOR})
+set(CPACK_PACKAGE_VERSION_MINOR ${VERSION_MINOR})
+set(CPACK_PACKAGE_VERSION_PATCH ${VERSION_PATCH})
+set(CPACK_INSTALL_DEFAULT_DIRECTORY_PERMISSIONS
+    OWNER_READ
+    OWNER_WRITE
+    OWNER_EXECUTE
+    GROUP_READ
+    GROUP_EXECUTE
+    WORLD_READ
+    WORLD_EXECUTE
+)
```

Comment:
Not sure if this is wise. Why not leave it at the default?

Reply:
I am actually restricting the rights compared with the automatically generated folders by the install commands in this CMakeLists.txt.
As the generated folders were having too lax permissions lintian was complaining.
I believe it is the same issues that happened here: https://gitlab.kitware.com/cmake/cmake/-/issues/17333

But you are right, the more accurate approach would be to set the folder rights at the file install time.

Reply:
**Update:**
After a little research it seems CMAKE_INSTALL_DEFAULT_DIRECTORY_PERMISSIONS and CPACK_INSTALL_DEFAULT_DIRECTORY_PERMISSIONS are the only way to set permissions for directories implicitly created at install time.
**Next Acions**
I can change the way to set it to CMAKE_INSTALL_DEFAULT_DIRECTORY_PERMISSIONS though to be completely identical between install and package target.